include/deal.II/grid/tria_accessor.h

```diff
@@ -3858,7 +3858,10 @@ class CellAccessor : public TriaAccessor<dim, dim, spacedim>
    */
 
   /**
-   * Return the orientation of this cell.
+   * Return the orientation of this cell. This function always returns
+   * `true` if `dim==spacedim`. It can return `true` or `false` if
+   * `dim==spacedim-1`. The function cannot be called (and will throw
+   * an exception) if called for `dim<spacedim-1`.
```

Comment:
We are calling `Assert` so we would normally abort instead of throwing an exception.

Reply:
We use this same phrase in many many places in the documentation even in places where we use `Assert` internally. But you're right, this shouldn't mean that we should continue this practice. So I fixed this in both places.